### ISSUE TITLE
Pretrained himtok.pth can't reconstruct the mask

### ISSUE THREAD

**Issue by SearchNova** [yayafengzi/LMM-HiMTok#10]
Thank you for sharing this great work!
I'm trying to use the pretrained himtok.pth model to encode a binary mask into tokens and then decode it back to reconstruct the mask. However, I noticed that the reconstructed mask is not identical to the original input.

`if __name__ == "__main__":
    himt_path = "./outputs/himtok.pth"
    himt = MaskDecoder.init_model_from_config( 
            model_path=himt_path,
            config_path="./config/himt.yaml",
            need_encoder=True,
            need_decoder=True,
        )
    himt.eval().cuda()

    mask = Image.open("./example/masks/1.png")
    img_size = mask.size
    mask = mask.convert("L").resize((256, 256))
    input_mask = torch.tensor(np.array(mask)).unsqueeze(0)
    input_mask = (input_mask.float()/255).cuda()
    tokens = himt.encode_mask(input_mask)
    print(tokens.shape)
    decoder_img = himt.decode_tokens(tokens)
    print(decoder_img.shape)
    str_tokens = SEG_START_TOKEN + "".join([SEG_TOKEN_TEMPLATE.format(token) for token in tokens[0]]) + SEG_END_TOKEN
      # (width, height)
    target_size = (img_size[0], img_size[1])  
    mask = decoder_img.squeeze(0).squeeze(0)
    mask = ((mask.float().detach().cpu().numpy()>0.5)*255).astype(np.uint8)
    mask = Image.fromarray(mask).resize(target_size)
    mask.save('./results/1.png')
    print(str_tokens)`
Looking forward to your reply！

**Comment by Labhahaha** [comment]
Thank you for sharing this excellent work!
I am encountering the same issue as the user above. I used the pre-trained himtok.pth model to encode binary masks into tokens and then decode them to reconstruct the masks. When I evaluated this on the ReferCOCO dataset series, the reconstruction performance was unsatisfactory.
Specifically, using 32 tokens, I only achieved an average reconstruction IoU of around 50%, and the reconstructed mask is visibly different from the input mask.
Could you please give me some advice or share the correct inference implementation? I would really appreciate your help!
Looking forward to your reply!
